solidity/test/maintainer/Maintainer.test.ts

```diff
@@ -0,0 +1,3579 @@
+        await restoreSnapshot()
+      })
+
+      context("when the challenge exists", () => {
+        context("when the challenge is open", () => {
+          context("when the heartbeat message has correct format", () => {
+            const heartbeatMessage = "0xFFFFFFFFFFFFFFFF0000000000E0EED7"
+            const heartbeatMessageSha256 = sha256(heartbeatMessage)
+            const sighash = sha256(sha256(heartbeatMessage))
+
+            let tx: ContractTransaction
+            let initThirdPartyBalance: BigNumber
+
+            before(async () => {
+              await createSnapshot()
+
+              const signature = ethers.utils.splitSignature(
+                heartbeatWalletSigningKey.signDigest(sighash)
+              )
+
+              await bridge
+                .connect(thirdParty)
+                .submitFraudChallenge(
+                  heartbeatWalletPublicKey,
+                  heartbeatMessageSha256,
+                  signature,
+                  {
+                    value: fraudChallengeDepositAmount,
+                  }
+                )
+
+              initThirdPartyBalance = await provider.getBalance(
+                thirdParty.address
+              )
+              tx = await maintainerProxy
+                .connect(thirdParty)
+                .defeatFraudChallengeWithHeartbeat(
+                  heartbeatWalletPublicKey,
+                  heartbeatMessage
+                )
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            it("should not revert", async () => {
+              await expect(tx.wait()).not.to.be.reverted
+            })
+
+            it("should refund ETH", async () => {
+              const postNotifyThirdPartyBalance = await provider.getBalance(
+                thirdParty.address
+              )
+              const diff = postNotifyThirdPartyBalance.sub(
+                initThirdPartyBalance
+              )
+
+              expect(diff).to.be.gt(0)
+              expect(diff).to.be.lt(
+                ethers.utils.parseUnits("1000000", "gwei") // 0,001 ETH
+              )
+            })
+          })
+        })
+      })
+    })
+  })
+
+  describe("authorize", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          maintainerProxy
+            .connect(thirdParty)
+            .authorize(thirdPartyContract.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the owner", () => {
+      let tx: ContractTransaction
+
+      before(async () => {
+        await createSnapshot()
+
+        tx = await maintainerProxy
+          .connect(governance)
+          .authorize(thirdPartyContract.address)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should authorize a contract", async () => {
+        await expect(
```

Comment:
The first `await` is not needed.

Reply:
[695eac7](https://github.com/keep-network/tbtc-v2/pull/209/commits/695eac72c524c2877954e301744f09ed2a92bebd)